File: C++/Ohne Polymorphismus/Haustier.cpp
#include "Haustier.h"

void Haustier::schreibeName()
{
	cout << name << endl;
}

Haustier::Haustier(HaustierArt typ, string name)
{
	this->type = typ;
	this->name = name;
}

void Haustier::fressen()
{
	switch (type)
	{
	case HaustierArt::HUND:
		cout << "Knochen nagen" << endl;
		break;
	case HaustierArt::KATZE:
		cout << "Maus fressen" << endl;
		break;
	case HaustierArt::WELLENSITTICH:
		cout << "Korn fressen" << endl;
		break;
	}
}

void Haustier::freuen()
{
	switch (type)
	{
	case HaustierArt::HUND:
		cout << "wedeln" << endl;
		break;
	case HaustierArt::KATZE:
		cout << "schnurren" << endl;
		break;
	case HaustierArt::WELLENSITTICH:
		cout << "wippen" << endl;
		break;
	}
}

void Haustier::lautGeben()
{
	switch (type)
	{
	case HaustierArt::HUND:
		cout << "bellen" << endl;
		break;
	case HaustierArt::KATZE:
		cout << "miauen" << endl;
		break;
	case HaustierArt::WELLENSITTICH:
		cout << "zwitschern" << endl;
		break;
	}
}

File: TypeScript/Ohne Polymorphismus/Haustier.ts
import {HaustierArt} from "./HaustierArt";

export class Haustier{
    protected name : string;
    protected type : HaustierArt;

    constructor(a : HaustierArt, n : string){
        this.type = a;
        this.name = n;
    }

    fressen():void {
        switch (this.type) {
            case HaustierArt.HUND:
                console.log("Knochen nagen");
                break;
            case HaustierArt.KATZE:
                console.log("Maus fressen");
                break;
            case HaustierArt.WELLENSITTICH:
                console.log("Korn fressen");
                break;
        }
    }

    freuen():void {
        switch (this.type) {
            case HaustierArt.HUND:
                console.log("wedeln");
                break;
            case HaustierArt.KATZE:
                console.log("schnurren");
                break;
            case HaustierArt.WELLENSITTICH:
                console.log("wippen");
                break;
        }
    }

    lautGeben():void {
        switch (this.type) {
            case HaustierArt.HUND:
                console.log("bellen");
                break;
            case HaustierArt.KATZE:
                console.log("miauen");
                break;
            case HaustierArt.WELLENSITTICH:
                console.log("zwitschern");
                break;
        }
    }

    schreibeName(){
        console.log(this.name);
    }
}


